.github/workflows/alarm.yml

```diff
@@ -18,9 +19,319 @@ jobs:
+          WANDB_SILENT: "true"
+
+          # --- Parameters ---
+          MAX_REVISIONS: "5"
+          NO_CHANGE_PATIENCE: "100"
+          RUNTIME_REGRESSION_TOLERANCE_PCT: "10"
+          COMPILE_REGRESSION_TOLERANCE_PCT: "10"
+
+          # Input/Output paths
+          ARTIFACTS_DIR: ${{ steps.dl.outputs.download-path }}
+          PR_COMMENT_PATH: pr_comment.md
+        run: |
+          python - <<'PY'
+          import os, sys, json, math, re, requests
+          import statistics as stats
+          import wandb
+
+          SHA_RE = re.compile(r"[0-9a-fA-F]{7,40}")
+
+          def _norm_rev(text):
+              if not text: return None
+              text = text.split("@", 1)[0]
+              m = SHA_RE.search(text)
+              return m.group(0) if m else text
+
+          def _normalize_kv_id(kv: dict, drop_keys=None) -> str:
+              drop = set(drop_keys or [])
+              pairs = []
+              for k, v in kv.items():
+                  if k in drop or v is None: continue
+                  k = str(k).strip(); v = str(v).strip()
+                  if not k or not v: continue
+                  pairs.append((k, v))
+              pairs.sort(key=lambda x: x[0])
+              return "-".join(f"{k}={v}" for k, v in pairs)
+
+          def wandb_normalize_benchmark_id(bid: str) -> str:
+              kv = {}
+              for token in (p.strip() for p in bid.split("-") if p.strip()):
+                  if "=" not in token: continue
+                  k, v = token.split("=", 1)
+                  kv[k.strip()] = v.strip()
+              return _normalize_kv_id(kv)
+
+          def artifacts_parse_speed_txt_lines(lines):
+              METRIC_KEYS = {"compile_time", "runtime_fps", "realtime_factor"}
+              out = {}
+              for line in lines:
+                  if not line.strip(): continue
+                  parts = [p.strip() for p in line.strip().split("|") if "=" in p]
+                  kv = {}
+                  for p in parts:
+                      k, v = p.split("=", 1)
+                      kv[k.strip()] = v.strip()
+                  tid = _normalize_kv_id(kv, drop_keys=METRIC_KEYS)
+                  rt = kv.get("runtime_fps"); ct = kv.get("compile_time")
+                  try: rt = float(rt) if rt is not None else None
+                  except: rt = None
+                  try: ct = float(ct) if ct is not None else None
+                  except: ct = None
+                  out[tid] = {"runtime_fps": rt, "compile_time": ct}
+              return out
+
+          # ----- Read event / find PR -----
+          API = os.environ.get("GITHUB_API_URL", "https://api.github.com")
+          ev  = json.load(open(os.environ["GITHUB_EVENT_PATH"], "r", encoding="utf-8"))
+          wr  = ev["workflow_run"]
+          repo= ev["repository"]
+          owner, name = repo["full_name"].split("/", 1)
+          head_sha = wr.get("head_sha")
+          wr_id = wr["id"]
+
+          s = requests.Session()
```

Comment:
Alexis mentioned that we'd better not
```
use a Python script writing low-level HTTP requests to comment on PRs instead of using existing Github Actions to hide the complexity
```